book/src/ephemeral-local-volumes.md

```diff
@@ -77,6 +77,11 @@ A CSI driver is suitable for CSI ephemeral inline volumes if:
 - ephemeral inline volumes have to be supported on Kubernetes clusters
   which do not support generic ephemeral volumes
 
+A CSI driver is not suitable for CSI ephemeral inline volumes when:
+- provisioning is not local to the node
+- ephemeral volume creation requires administration-level parameters, for
```

Comment:
Another aspect is if the volumeAttributes contains information that would want to be restricted by an administrator (such as volume references when there's no Pod-level authorization)

Reply:
Can you give an example here?

My first thought was to say something like "provisioning is done in the controller and would need to be replicated in nodePublish", but that felt too functional. I think that's roughly the idea, though: if adding csi ephemeral support would require moving/duplicating provisioning code from the controller to the node then there's a bad smell.

But it would be good to be more precise I think.

Reply:
So with this feature, you can set the volumeAttributes directly in the Pod spec. So that's what I'm referring to here. If any of those attributes are things that an admin would want to restrict, then PV/StorageClass is a better place for it.

In this particular example, the driver has nfs server and share as attributes. Those are things an admin would want to restrict and not allow any Pod to specify anything.

Reply:
updated, ptal